packages/mermaid/src/diagrams/git/gitGraphParser.ts

```diff
@@ -0,0 +1,90 @@
+import type { GitGraph } from '@mermaid-js/parser';
+import { parse } from '@mermaid-js/parser';
+import type { ParserDefinition } from '../../diagram-api/types.js';
+import { log } from '../../logger.js';
+import { populateCommonDb } from '../common/populateCommonDb.js';
+import db from './gitGraphAst.js';
+import { commitType } from './gitGraphAst.js';
+import type {
+  CheckoutAst,
+  CherryPickingAst,
+  MergeAst,
+  CommitAst,
+  BranchAst,
+} from './gitGraphTypes.js';
+
+const populate = (ast: GitGraph) => {
+  populateCommonDb(ast, db);
+  // @ts-ignore: this wont exist if the direction is not specified
+  if (ast.dir) {
+    // @ts-ignore: this wont exist if the direction is not specified
+    db.setDirection(ast.dir);
+  }
+  for (const statement of ast.statements) {
+    parseStatement(statement);
+  }
+};
+
+const parseStatement = (statement: any) => {
+  switch (statement.$type) {
+    case 'Commit':
+      parseCommit(statement);
+      break;
+    case 'Branch':
+      parseBranch(statement);
+      break;
+    case 'Merge':
+      parseMerge(statement);
+      break;
+    case 'Checkout':
+      parseCheckout(statement);
+      break;
+    case 'CherryPicking':
+      parseCherryPicking(statement);
+      break;
+    default:
+      log.warn(`Unknown statement type`);
```

Comment:
Should we just log, or throw an error?

We should log an error in either case. 
```suggestion
    default:
      log.error(`Unknown statement type`);
```